content/blog/kubernetes-cluster-as-code-part-2.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Kubernetes Cluster as Code - Part 2
+date: 2022-06-23T14:01:05.923Z
+author: Rachana Kallada Jayaraj
+authorimage: /img/photo-1-.jpg
+tags:
+  - hpe-greenlake
+  - devops
+  - terraform
+  - open source
+---
+<!--StartFragment-->
+
+## Getting Started
+
+The process of managing and provisioning computer data centers through machine-readable definition files, also known as Infrastructure-as-Code (IaC), offers many significant benefits. It helps to increase operational agility, simplify management, reduce errors, and save cost. In this post, we will explore some of the benefits of using IaC to build a Kubernetes cluster from scratch, with all the necessary configuration and core services, on HPE GreenLake using Terraform (TF). Storing Kubernetes cluster and favorable configurations as code, helps in repeatability and change management.
+
+IaC with Kubernetes is not new. There are providers in the developer community which are quite good and well supported. Using the HPE GreenLake Terraform provider, you can bring up a Kubernetes cluster starting right from the infrastructure layer and way up in the stack, to set up desired configurations and applications. For reference, see the below picture.
+
+<!--EndFragment-->
+
+![](/img/image2022-6-20_12-36-56.png)
+
+<!--StartFragment-->
+
+HPE GreenLake TF provider brings the Kubernetes stack up on the HPE GreenLake Infrastructure, and exposes credentials for other TF providers to integrate further and build the complete stack, as desired. In the diagram above, 2 and 3 are community providers that are available, which can be used in combination with HPE GreenLake TF provider. 
+
+<br/>
+
+## Let's Recap
+
+In [](https://rndwiki-pro.its.hpecorp.net/display/HCSS/Blog+%3A+Kubernetes+clusters+as+Code+-+Part1)the Kubernetes Cluster as Code - Part 1 blog , we covered the usage of the HPE GreenLake Terraform provider:
+
+* To create and destroy a cluster resource.
+* To incorporate community providers like kubernetes and create namespaces in the created cluster.
+
+In this blog, we will focus on managing application deployments using IaC. Here, we will be deploying applications on an existing/pre-created cluster. Hence, the pre-requisite to proceed, would be, to have a cluster and a namespace created. You can follow the steps mentioned in Kubernetes Cluster as Code - Part 1 blog.
```

Comment:
You may want to tell the reader here that you are going to deploy Prometheus as an application on the K8s cluster.

Reply:
Yes, done